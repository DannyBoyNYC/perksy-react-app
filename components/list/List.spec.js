const React = require('react');
const { shallow } = require('enzyme');

const List = require('./index').default;

describe('<List.* />', () => {
  let component;
  const requiredProps = {
    items: ['foo', 'bar']
  };

  describe('<List.Unordered />', () => {
    beforeEach(() => {
      component = shallow(<List.Unordered {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('list list--unordered');
    });

    it('should render a <ul> tag', () => {
      expect(component).toMatchSelector('ul');
    });
  });

  describe('<List.Ordered />', () => {
    beforeEach(() => {
      component = shallow(<List.Ordered {...requiredProps} />).dive();
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should have the correct BEM class names', () => {
      expect(component).toHaveClassName('list list--ordered');
    });

    it('should render a <ol> tag', () => {
      expect(component).toMatchSelector('ol');
    });
  });
});
