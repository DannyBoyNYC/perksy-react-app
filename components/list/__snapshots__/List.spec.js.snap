// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List.* /> <List.Ordered /> should render correctly with all required props 1`] = `
<ol
  className="list list--ordered"
>
  <li
    className="list-item"
    key="0"
  >
    foo
  </li>
  <li
    className="list-item"
    key="1"
  >
    bar
  </li>
</ol>
`;

exports[`<List.* /> <List.Unordered /> should render correctly with all required props 1`] = `
<ul
  className="list list--unordered"
>
  <li
    className="list-item"
    key="0"
  >
    foo
  </li>
  <li
    className="list-item"
    key="1"
  >
    bar
  </li>
</ul>
`;
